src/System.Security.Principal.Windows/src/System/Security/Principal/NTAccount.cs

```diff
@@ -142,19 +142,7 @@ public override IdentityReference Translate(Type targetType)
 
         public override bool Equals(object o)
         {
-            if (o == null)
-            {
-                return false;
-            }
-
-            NTAccount nta = o as NTAccount;
-
-            if (nta == null)
-            {
-                return false;
-            }
-
-            return (this == nta); // invokes operator==
+            return (this == o as NTAccount); // invokes operator==
```

Comment:
Personally, I'd have parentheses around `o as NTAccount`, or leave it as a local variable; but that's just because my brain-parser is pivoting on the syntax-highlighted word "as". (I know it's not ambiguous to the compiler :)).

Certainly not a required change, but if there was another iteration for any reason it'd give me a bit of a warm fuzzy feeling.